Accept InstanceDistribution with only one instance type.

ryotarai opened the pull request:
### Description

Currently, InstanceDistribution must have more than or equal to 2 instance types but AWS auto scaling API accepts configuration with only one instance type too. This PR makes InstanceDistribution accept that case.

#### Q. Why we do not use just `instanceType` of nodegroup?

We want to launch spot instances with one instance type.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested
  - To run tests, https://github.com/weaveworks/eksctl/pull/1771 is needed
- [ ] Added labels for change area (e.g. `area/nodegroup`) and target version (e.g. `version/0.12.0`)
- [ ] Added note in `docs/release_notes/draft.md` (or relevant release note)

<!-- If you haven't done so already, you can add your name to the humans.txt file -->
<!-- If you need the attention of the maintainers ping @weaveworks/eksctl -->

ryotarai commented:
@martina-if Hi, would you please this PR?

martina-if commented:
Hi @ryotarai , last time I checked this wasn't supported with MixedInstances, do you have a link to some docs where it says now this is supported? (it was supported through another mechanism though https://github.com/weaveworks/eksctl/pull/781/files)

ryotarai commented:
Thank you for reviewing.

> last time I checked this wasn't supported with MixedInstances, do you have a link to some docs where it says now this is supported?

https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_LaunchTemplate.html
This page says `You can specify between 1 and 20 instance types.` and the following config worked in my environment.

```yaml
  instancesDistribution:
    instanceTypes: ["c5.large"]
    spotInstancePools: 1
    onDemandBaseCapacity: 0
    onDemandPercentageAboveBaseCapacity: 0
```

> it was supported through another mechanism though https://github.com/weaveworks/eksctl/pull/781/files

I didn't notice that. Thank you.

martina-if commented:
This is great! Apparently they changed that [in October](https://github.com/awsdocs/aws-cloudformation-user-guide/commit/90f137b2726547b565405c6db4f1a4764f16f3a5#diff-25d2d609062098fc251ceb3655ef83d4).

Ok, two tiny details to complete the PR:
  - [ ] https://github.com/weaveworks/eksctl/blob/master/examples/08-spot-instances.yaml#L16 
  - [ ] https://github.com/weaveworks/eksctl/blob/master/site/content/usage/08-spot-instances.md

Can you please fix the comment in those two files^? It says minimum 2 instance types. Then if you rebase I'll approve and merge it :)

Thanks a lot!

ryotarai commented:
I've updated the example and doc and rebased this to master branch: https://github.com/weaveworks/eksctl/pull/1772/commits/5d9436dc8faf5f35eee6f4b14a685d4728500df2
(Should I squash commits into one commit?)

martina-if commented:
Hi @ryotarai , yes it would be great if you could squash them :)

ryotarai commented:
I've squashed commits into https://github.com/weaveworks/eksctl/pull/1772/commits/2cc4936407690e8cfd382be1aef2554cc6c14466

martina-if commented:
Thank you @ryotarai !
